Remove workaround for Homebrew libc++ issue

mppf opened the pull request:
This PR removes a workaround for issue #19217 (which is the same issue as https://github.com/Homebrew/homebrew-core/issues/96915 and https://github.com/llvm/llvm-project/issues/54438 ).

The problem is solved in the current Homebrew LLVM 14 and LLVM 15 formulae now that libc++ is installed into a different directory than the LLVM libraries. (This problem would still be present for the LLVM 12 and 13 formulae). Since the workaround is removed, this PR adjusts printchplenv to raise an error if using Mac OS X with system LLVM 11, 12, or 13. Such LLVMs will not be selected as a quickstart configuration. Also, it updates prereqs.rst to indicate that LLVM 14 or newer is required on Mac OS X.

Reviewed by @daviditen - thanks!

- [x] testing on Mac OS X 10.15 "Catalina"
- [x] testing on Mac OS X 11 "Big Sur"
- [x] testing on Mac OS X 12 "Monterey" (x86)
- [x] testing on Mac OS X 12 "Monterey" (M1)

daviditen reviewed (approved):
I think this will need a documentation update to say that only LLVM-14 is supported for homebrew on OSX. That can come as a separate PR.

mppf commented:
@daviditen - I've added an error for LLVM 11/12/13 on Mac OS X and also updated the prereqs.rst. Could you check the last two commits to make sure you are happy with these changes? Thanks.

daviditen commented:
Yes, the new changes LGTM, thanks!
